### ISSUE TITLE
Owncloud - allows self signed certificates

### ISSUE THREAD

**Issue by SpamReceiver** [mendhak/gpslogger#376]
I have an Owncloud instance running in my home network. This server is not reachable from the internet, only within my LAN or via an VPN tunnel. So, even if the phone has an internet connection, the server may be unreachable. The app should cater for this situation and stall the upload(s) until the server becomes reachable again.

Second issue with my Owncloud instance is that it is running on a NAS, which has a built-in certificate. This certificate cannot be verified against any trusted CA. Thus, the app does not allow to connect to my Owncloud instance. The app should present the user the server's certificate and let the user decide whether to trust or not to trust it.

**Comment by mendhak** [comment]
The closest would be to enable 'Send on Wi-Fi only' option in upload settings.  Anything beyond that, such as waiting for the server to return would require unknown amount of storage space as well as to keep issuing requests to the target domain - but the target domain may return after several months or years.  

Second issue is around self signed certs.  I have added some code to try and accept self signed certs.  APK is attached with this post.  If you can test it and say that it works, that would be great.

[gpslogger-70-issue376.apk.zip](https://github.com/mendhak/gpslogger/files/102385/gpslogger-70-issue376.apk.zip)

**Comment by SpamReceiver** [comment]
I have tried the updated version gpslogger-70-issue376 with no success. I have captured the attached log output and hope this helps you with an analysis.

[debuglog.txt](https://github.com/mendhak/gpslogger/files/104869/debuglog.txt)

**Comment by mendhak** [comment]
That debug log did help but this issue was very difficult to figure out. :crying_cat_face: 

I have taken the OwnCloud library and [modified it](https://github.com/mendhak/owncloud-android-library/commit/52bff6280ddae92e970b51447f74ee8935900117) to allow for Self Signed SSL certs.

Try this APK:

https://www.dropbox.com/s/ggzjljgau4ltdhj/gpslogger-70-issue376-2.apk?dl=0


For my setup, I used [this docker](https://hub.docker.com/r/jchaney/owncloud/) with `make owncloud-https`.  I then generated a cert using `make-ssl-cert generate-default-snakeoil` which hopefully generated as self signed as a user would get I hope.

With this setup I was able to get the same error as you.  

After hunting and finally making the tiny change, it seems to work.  

I'll have to submit a pull request to get this upstream if it's to be included in the official repo.
